### PR TITLE
Expand ozzie.yaml to fail the build based on warnings and/or errors

### PR DESCRIPTION
**Pull request by jorgecoca**

### CODE REVIEW

File: lib/exceptions/fail_build_exception.dart [hunk 1]
```diff
@@ -0,0 +1,33 @@
+/// Exception thrown when Ozzie fails a set of integration tests
+class FailBuildException implements Exception {
+  String message;
+
+  FailBuildException(this.message);
+
+  /// Helper to build an exception based on warnings in the report
+  factory FailBuildException.onWarning(String featureName) {
+    return FailBuildException("""
+¯\_(ツ)_/¯  Warnign detected
```
Comment (felangel) [inline lib/exceptions/fail_build_exception.dart]: typo: `Warnign`

File: lib/exceptions/fail_build_exception.dart [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+Ozzie has measured warnings on the $featureName feature, and your configuration
```
Comment (felangel) [inline lib/exceptions/fail_build_exception.dart]: nit but might want to reword to `Ozzie has **detected** warnings...`
Also typo on the next like `warnigns`

File: lib/exceptions/fail_build_exception.dart [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+Check your configuration at ozzie.yaml in case you want to make adjustments.
+  """);
+  }
+
+  /// Helper to build an exception based on errors in the report
+  factory FailBuildException.onError(String featureName) {
+    return FailBuildException("""
+(╯°□°)╯︵ ɹoɹɹƎ Detected
+
+Ozzie has measured errors on the $featureName feature, and your configuration
```
Comment (felangel) [inline lib/exceptions/fail_build_exception.dart]: same here might want to reword to `Ozzie has **detected** errors...`

File: lib/exceptions/fail_build_exception.dart (hunk 2 above)
Comment (jorgecoca) [reply lib/exceptions/fail_build_exception.dart]: 🤦‍♂

File: test/exceptions/fail_build_exception_test.dart [hunk 4]
```diff
@@ -0,0 +1,45 @@
+import 'package:test/test.dart';
+import 'package:ozzie/exceptions/exceptions.dart';
+
+void main() {
+  group('FailBuildException', () {
+    test('onWarning message is properly formatted', () {
+      final e = FailBuildException.onWarning('test123');
+      final expectedMessage = """
+¯\_(ツ)_/¯  Warnign detected
```
Comment (felangel) [inline test/exceptions/fail_build_exception_test.dart]: same typo

**Pull request merged**